storage/src/objects/dpc_state.rs

```diff
@@ -128,15 +128,23 @@ impl<T: Transaction, P: LoadableMerkleParameters, S: Storage> Ledger<T, P, S> {
     }
 
     /// Rebuild the stored merkle tree with the current stored commitments
-    pub fn update_merkle_tree(&self) -> Result<(), StorageError> {
+    pub fn update_merkle_tree(&self, new_best_block_number: u32) -> Result<(), StorageError> {
         self.rebuild_merkle_tree(vec![])?;
+        let new_digest = self.cm_merkle_tree.read().root();
 
-        let update_current_digest = DatabaseTransaction(vec![Op::Insert {
+        let mut database_transaction = DatabaseTransaction::new();
+
+        database_transaction.push(Op::Insert {
```

Comment:
Not entirely sure if this operation is necessary. Because we only need to insert into `COL_DIGEST` when a new block is committed and remove when a block is decommitted.

Reply:
Without this additional operation, the `consensus_sidechain::{fork_out_of_order, new_out_of_order}` fail due to a `Ledger digest is invalid.` error ([CI logs](https://app.circleci.com/pipelines/github/AleoHQ/snarkOS/2606/workflows/a21ff086-f4c7-4bb9-beca-a2b180f67363/jobs/6439)).

Reply:
Unless one of the 2 currently inactive operations:
```
database_transaction.push(Op::Delete {
    col: COL_DIGEST,
    key: self.current_digest()?,
});
```
should just be deleted - then this change is not needed.